### PR TITLE
Fix sending server up commands from workers

### PR DESCRIPTION
**Pull request by erikjohnston**
Add fix the typing of `synapse.federation.transport`

### CODE REVIEW

File: changelog.d/6811.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix waking waking up other servers when remote server is detected to have come back online from other workers.
```
Comment (anoadragon453) [inline changelog.d/6811.bugfix]:
```suggestion
Fix waking up other workers when remote server is detected to have come back online.
```

File: synapse/federation/transport/client.py [hunk 2]
```diff
@@ -352,7 +353,9 @@ def get_public_rooms(
         else:
             path = _create_v1_path("/publicRooms")
 
-            args = {"include_all_networks": "true" if include_all_networks else "false"}
+            args = {
+                "include_all_networks": "true" if include_all_networks else "false"
+            }  # type: Dict[str, Any]
```
Comment (anoadragon453) [inline synapse/federation/transport/client.py]:
```suggestion
            }  # type: Dict[str, str]
```

?

File: synapse/federation/transport/server.py [hunk 3]
```diff
@@ -267,6 +268,8 @@ class BaseFederationServlet(object):
                 returned.
     """
 
+    PATH = ""  # Overriden in subclasses, the regex to match against the path.
```
Comment (anoadragon453) [inline synapse/federation/transport/server.py]:
```suggestion
    PATH = ""  # Overridden in subclasses, the regex to match against the path.
```

File: synapse/federation/transport/server.py [hunk 4]
```diff
@@ -824,7 +824,7 @@ def __init__(self, handler, authenticator, ratelimiter, server_name, allow_acces
         if not self.allow_access:
             raise FederationDeniedError(origin)
 
-        limit = int(content.get("limit", 100))
+        limit = int(content.get("limit", 100))  # type: Optional[int]
```
Comment (anoadragon453) [inline synapse/federation/transport/server.py]: Will `limit` ever be `None`? `int(None)` raises a `TypeError`.

**Comment by erikjohnston** [comment]
> I see a lot of fixes but not actually sure what broke waking up other workers about a server's status?

Sorry, its https://github.com/matrix-org/synapse/pull/6811/commits/f2c8ef8976b3905ecc633f46099dfc076c2fbc7c which i discovered during mypy'ing

File: synapse/federation/transport/client.py (hunk 2 above)
Comment (erikjohnston) [reply synapse/federation/transport/client.py]: We end up setting the values to tuples and lists below.

File: synapse/federation/transport/server.py (hunk 4 above)
Comment (erikjohnston) [reply synapse/federation/transport/server.py]: Limit shouldn't be none here, but later we'll set it to `None` if its zero

File: synapse/federation/transport/client.py (hunk 2 above)
Comment (anoadragon453) [reply synapse/federation/transport/client.py]: Ah, yes we do.

**Pull request merged**